packages/x-data-grid-premium/src/components/GridFooterCell.tsx

```diff
@@ -54,6 +57,7 @@ function GridFooterCell(props: GridFooterCellProps) {
   const classes = useUtilityClasses(ownerState);
 
   return (
+    // @ts-ignore `@mui/material` system styled() doesn't have a compatible sx prop
     <GridFooterCellRoot ownerState={ownerState} className={classes.root} {...other}>
```

Comment:
Using system `styled()` for `GridFooterCellRoot` means it's incompatible because we explicitly declare the type as `sx?: MaterialTheme`.

Reply:
I think this should be fixed by my suggestion in `internals/styled`:

```js
createStyled<Theme>({ themeId: MATERIAL_THEME_ID });
```